compiler/mir/mirconstr.nim

```diff
@@ -45,6 +143,188 @@ template follows*(): Sink =
+  result = EValue(typ: n.typ)
+
+func constr*(s: var MirNodeSeq, typ: PType): EValue =
+  s.add MirNode(kind: mnkConstr, typ: typ)
+  result = EValue(typ: typ)
+
+func temp*(s: var MirNodeSeq, typ: PType, id: TempId): EValue =
+  s.add MirNode(kind: mnkTemp, typ: typ, temp: id)
+  result = EValue(typ: typ)
+
+func symbol*(s: var MirNodeSeq, kind: range[mnkConst..mnkLocal],
+             sym: PSym): EValue =
+  s.add MirNode(kind: kind, typ: sym.typ, sym: sym)
+  result = EValue(typ: sym.typ)
+
+func opParam*(s: var MirNodeSeq, i: uint32, typ: PType): EValue =
+  assert typ != nil
+  s.add MirNode(kind: mnkOpParam, param: i, typ: typ)
+  result = EValue(typ: typ)
+
+func magicCall*(s: var MirNodeSeq, m: TMagic, typ: PType): EValue =
+  assert typ != nil
+  s.add MirNode(kind: mnkMagic, typ: typ, magic: m)
+  result = EValue(typ: typ)
+
+# input/output:
+
+func tag*(s: var MirNodeSeq, effect: EffectKind, val: var EValue) =
+  s.add MirNode(kind: mnkTag, effect: effect, typ: val.typ)
+
+func castOp*(s: var MirNodeSeq, typ: PType, val: var EValue) =
+  s.add MirNode(kind: mnkCast, typ: typ)
+  val.typ = typ
+
+func stdConvOp*(s: var MirNodeSeq, typ: PType, val: var EValue) =
+  s.add MirNode(kind: mnkStdConv, typ: typ)
+  val.typ = typ
+
+func convOp*(s: var MirNodeSeq, typ: PType, val: var EValue) =
+  s.add MirNode(kind: mnkConv, typ: typ)
+  val.typ = typ
+
+func addrOp*(s: var MirNodeSeq, typ: PType, val: var EValue) =
+  s.add MirNode(kind: mnkAddr, typ: typ)
+  val.typ = typ
+
+func viewOp*(s: var MirNodeSeq, typ: PType, val: var EValue) =
+  s.add MirNode(kind: mnkView, typ: typ)
+  val.typ = typ
+
+func derefOp*(s: var MirNodeSeq, typ: PType, val: var EValue) =
+  s.add MirNode(kind: mnkDeref, typ: typ)
+  val.typ = typ
+
+func derefViewOp*(s: var MirNodeSeq, typ: PType, val: var EValue) =
+  s.add MirNode(kind: mnkDerefView, typ: typ)
+  val.typ = typ
+
+func pathObj*(s: var MirNodeSeq, field: PSym, val: var EValue) =
+  assert field.kind == skField
+  s.add MirNode(kind: mnkPathNamed, typ: field.typ, field: field)
+  val.typ = field.typ
+
+func pathPos*(s: var MirNodeSeq, elemType: PType, position: uint32, val: var EValue) =
+  s.add MirNode(kind: mnkPathPos, typ: elemType, position: position)
+  val.typ = elemType
+
+func pathVariant*(s: var MirNodeSeq, objType: PType, field: PSym, val: var EValue) =
+  s.add MirNode(kind: mnkPathVariant, typ: objType, field: field)
+  val.typ = objType
+
+func unaryMagicCall*(s: var MirNodeSeq, m: TMagic, typ: PType, val: var EValue) =
+  assert typ != nil
+  s.add MirNode(kind: mnkMagic, typ: typ, magic: m)
+  val.typ = typ
+
+# sinks:
+
+func arg*(s: var MirNodeSeq, val: EValue) =
+  s.add MirNode(kind: mnkArg, typ: val.typ)
+
+func consume*(s: var MirNodeSeq, val: EValue) =
+  s.add MirNode(kind: mnkConsume, typ: val.typ)
+
+func name*(s: var MirNodeSeq, val: EValue) =
+  s.add MirNode(kind: mnkName, typ: val.typ)
+
+func voidOut*(s: var MirNodeSeq, val: EValue) =
```

Comment:
```suggestion
func voidOut*(s: var MirNodeSeq, val: EValue) =
  ## end the mir input expression as a void-expression
```

not sure I have that quite right, this is from what I could gather based on usage. initially I was seeing if I could come up with a better name, settled on a doc string might help. (soft suggestion)

Reply:
Yup, that sounds correct. I added it plus some additional elaboration.

In general, the 'void' operator/node acts as both a `discard` and as a delimiter for expressions ('call' operators that yield nothing also need to be followed by 'void').